File: training_camps/trainingcamp2022/2/contest/E.cpp
#include <bits/stdc++.h>
using namespace std;
int main(){
		int s;
		string word,pok;
		cin>>s;
		cin>>word;
		map<string,int> pokesd;
		pokesd["vaporeon"]=0;
		pokesd["jolteon"]=0;
		pokesd["flareon"]=0;
		pokesd["espeon"]=0;
		pokesd["umbreon"]=0;
		pokesd["leafeon"]=0;
		pokesd["glaceon"]=0;
		pokesd["sylveon"]=0;
		
    vector<string> words ={"vaporeon", "jolteon", "flareon", "espeon", "umbreon", "leafeon", "glaceon", "sylveon"};
    for(int i=0;i<words.size();i++){
    	if(s==words[i].length()){
    		pok=words[i];
    		for(int ii=0;ii<words[i].length();i++){
    				if(word[ii]==words[i][ii]){
    					pokesd[words[i]]=pokesd[words[i]]+1;
    				}
    				if(ii>=2){
    					pok="";
    				}
    			}
    		}
    	}
    
    //map <string,float> :: iterator iter;

}

File: training_camps/trainingcamp2022/2/contest/E.py
input()
name=input()
pokes=["vaporeon", "jolteon", "flareon", "espeon", "umbreon", "leafeon", "glaceon", "sylveon"]
pokesd={"vaporeon":0, "jolteon":0, "flareon":0, "espeon":0, "umbreon":0, "leafeon":0, "glaceon":0, "sylveon":0}
pokeocurrence=[]
#if name==""
onlysize=False
for i in pokes:
	ocur=0
	if len(name)==len(i):
		for ii in range(len(name)):
			if i[ii]==name[ii]:
				onlysize=True
				ocur+=1
		pokesd[i]=ocur
	pokeocurrence.append(ocur)





	"""ocur=0
	for ii in name:
		if ii in i:
			print(i,name)
			for iii in range(len(name)):
				if i[iii]==name[iii]:
					pokesd[i]+=1
			ocur+=1
	"""
maxx=0
pos=0
poket=""
for j in range(len(pokeocurrence)):
	if maxx<pokeocurrence[j]:
		#print(pokesd)
		maxx=pokesd[pokes[j]]
		pos=j
		poket=pokes[j]
		#maxx=pokeocurrence[j]


#print(pokes[pos])
#print(pokesd)
if onlysize:
	print(poket)
else:
	ss=list(map(len,pokes))
	#ss.count(len(name))==
	if ss.count(len(name))==1:
		for i in range(len(pokes)):
			if len(pokes[i])==len(name):
				print(pokes[i])